### DISCUSSION TITLE
Question: Input delay related to it being an app?

### DISCUSSION THREAD

**Discussion by Yaw-Dev** [cafali/SnapKey#24] [Q&A]
As far as I'm aware the Snap Tap feature Razer introduced runs on a hardware level and so doesn't affect input latency at all. SnapKey however is a usermode application so in my mind latency could be added. Is that true?

**Comment by cafali** [comment]
I am not sure if  the new Razer feature is hardware-based, as it requires you to install external software to activate it. If it were functioning at the hardware level, a new key or shortcut could be introduced to trigger it directly from the keyboard.

The only distinction is that it is optimized for the new ultra-fast switches included in their latest keyboards, which helps reduce input latency. SnapKey, on the other hand, is designed to work with any keyboard, even the most basic $5 keyboards.

Regarding input latency, I personally haven’t noticed any delay while using it, and I lack the means to conduct formal latency benchmarks.

**Comment by Yaw-Dev** [comment]
After some testing I didn't observe anything noticable either. Thanks for this by the way, very cool project.
